Extension/src/LanguageServer/client.ts

```diff
@@ -868,6 +877,91 @@ export class DefaultClient implements Client {
+        const paths: string[] = compilerDefaults.knownCompilers.map(function (a: configs.KnownCompiler): string { return a.path; });
+        paths.push("Select another compiler on my machine");
+        paths.push("Help me install a compiler");
+        paths.push("Do not configure a compiler (not recommended)");
+        const index: number = await ui.showSelectCompiler(paths);
+        if (index === paths.length - 1) {
+            paths[index] = "";
+        } else if (index === paths.length - 2) {
+            switch (os.platform()) {
+                case 'win32':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217614");
+                    return;
+                case 'darwin':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217706");
+                    return;
+                case 'linux':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217615");
+                    return;
+                default:
+                    throw new Error(localize("unexpected.os", "Unexpected OS"));
+            }
+        } else if (index === paths.length - 3) {
+            const result: vscode.Uri[] | undefined = await vscode.window.showOpenDialog();
+            if (result !== undefined && result.length > 0) {
+                util.addTrustedCompiler(result[0].fsPath);
+                settings.defaultCompiler = result[0].fsPath;
+                const selectedCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);
+                compilerDefaults = selectedCompilerDefaults;
+                this.updateClientConfigurations();
+                return;
+            }
+        } else {
+            util.addTrustedCompiler(paths[index]);
+        }
+        settings.defaultCompiler = paths[index];
+        const selectedCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);
+        compilerDefaults = selectedCompilerDefaults;
+        this.updateClientConfigurations();
+
+    }
+
+    async promptSelectCompiler(command: boolean): Promise<void> {
+        const selectCompiler: string = localize("selectCompiler.string", "Select Compiler");
+        const confirmCompiler: string = localize("confirmCompiler.string", "Yes");
+        const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);
+        const settings: OtherSettings = new OtherSettings();
+        if (compilerDefaults.compilerPath !== "") {
+            if (!command) {
+                const value: string | undefined = await vscode.window.showInformationMessage(localize("selectCompiler.message", "The ${compilerDefaults.compilerPath} was found on this computer. Do you want to configure it for IntelliSense?"), confirmCompiler, selectCompiler);
```

Comment:
Should the message say "The compiler <compiler_path> was found on this computer." Instead of "The <compiler_path> .... "